crates/api/src/local_agent_store.rs

```diff
@@ -17,6 +17,9 @@ pub trait LocalAgentStore: 'static + Send + Sync + std::fmt::Debug {
 
     /// Get a list of all local agents currently in the store.
     fn get_all(&self) -> BoxFut<'_, K2Result<Vec<DynLocalAgent>>>;
+
+    /// Check whether an agent is local.
+    fn is_agent_local(&self, agent: id::AgentId) -> BoxFut<'_, K2Result<bool>>;
```

Comment:
I think breaking the API is preferable to always having to `get_all` and look for the agent in a vector when it's internally a hash map

Reply:
Totally, break the API as much as you want while the iron is still hot.